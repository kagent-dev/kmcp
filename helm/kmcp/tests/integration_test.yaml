suite: Integration tests for KMCP helm chart
templates:
  - deployment.yaml

tests:
  - it: should create deployment with default values
    set:
      image.repository: test-repo
      image.tag: v1.0.0
      controller.metrics.enabled: true
      serviceAccount.create: true
      rbac.create: true
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should create deployment with minimal configuration
    set:
      image.repository: test-repo
      image.tag: v1.0.0
      controller.metrics.enabled: false
      serviceAccount.create: false
      rbac.create: false
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should use custom namespace
    set:
      image.repository: test-repo
      image.tag: v1.0.0
      controller.metrics.enabled: true
      serviceAccount.create: true
      rbac.create: true
      global:
        namespaceOverride: custom-namespace
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should use custom release name
    set:
      image.repository: test-repo
      image.tag: v1.0.0
      controller.metrics.enabled: true
      serviceAccount.create: true
      rbac.create: true
      nameOverride: custom-release
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should include metrics port in deployment when metrics enabled
    set:
      image.repository: test-repo
      image.tag: v1.0.0
      controller.metrics.enabled: true
      controller.metrics.bindAddress: ":8443"
      serviceAccount.create: true
      rbac.create: true
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should create deployment when metrics enabled
    set:
      image.repository: test-repo
      image.tag: v1.0.0
      controller.metrics.enabled: true
      serviceAccount.create: true
      rbac.create: true
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {} 