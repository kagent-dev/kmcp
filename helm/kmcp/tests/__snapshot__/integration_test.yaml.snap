should create deployment when metrics enabled:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should create deployment with default values:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should create deployment with minimal configuration:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: default
          terminationGracePeriodSeconds: 10
          volumes: []
should include metrics port in deployment when metrics enabled:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should use custom namespace:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should use custom release name:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
